-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

- From 10d2ac1cb954f7a1ae7d66ceca8c6716618f675b Mon Sep 17 00:00:00 2001
From: Sebastian Bach <sebtomba@gmail.com>
Date: Fri, 26 May 2023 13:33:45 +0200
Subject: [PATCH] [PM-6226] Improve error messages

- ---
 ledger/src/transaction.rs   | 4 ++--
 pallets/midnight/src/lib.rs | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/ledger/src/transaction.rs b/ledger/src/transaction.rs
index 4988362f..31fa22cc 100644
- --- a/ledger/src/transaction.rs
+++ b/ledger/src/transaction.rs
@@ -21,8 +21,8 @@ pub struct Transaction(Tx<Proof>);
 
 impl Transaction {
 	pub fn from_bytes(encoded: &[u8]) -> Result<Self, String> {
- -		let tx: &str = std::str::from_utf8(encoded).unwrap();
- -		let tx: Tx<Proof> = from_str(tx).map_err(|e| e.to_string())?;
+		let json: &str = std::str::from_utf8(encoded).map_err(|e| e.to_string())?;
+		let tx: Tx<Proof> = from_str(json).map_err(|e| e.to_string())?;
 		Ok(Self(tx))
 	}
 
diff --git a/pallets/midnight/src/lib.rs b/pallets/midnight/src/lib.rs
index b14e66ec..dfc80c5b 100644
- --- a/pallets/midnight/src/lib.rs
+++ b/pallets/midnight/src/lib.rs
@@ -80,13 +80,13 @@ pub mod pallet {
 			}?;
 
 			let tx = ledger::Transaction::from_bytes(&encoded_transaction).map_err(|e| {
- -				log::error!("Can't deserialize the transaction: {}", e);
+				log::error!("❌ Can't deserialize the transaction: {}", e);
 				DispatchError::Other("Can't deserialize the transaction")
 			})?;
 			let new_state = state
 				.apply_transaction(&tx)
 				.map_err(|e| {
- -					log::error!("Can't apply the transaction: {}", e);
+					log::error!("❌ Can't apply the transaction: {}", e);
 					DispatchError::Other("Can't apply the transaction")
 				})?
 				.to_bytes();
@@ -124,7 +124,7 @@ pub mod pallet {
 				Some(state) => match ledger::Ledger::from_bytes(&state) {
 					Ok(l) => l.get_contract_state(contract_address).unwrap_or(Vec::new()),
 					Err(e) => {
- -						log::error!("Can't deserialize the Midnight Ledger state: {}", e);
+						log::error!("❌ Can't deserialize the Midnight Ledger state: {}", e);
 						Vec::new()
 					},
 				},
- -- 
2.46.0

-----BEGIN PGP SIGNATURE-----

iQIzBAEBCAAdFiEE9Oz16eF+Nv5C7TpSdEmbFI1hGmwFAmhtArIACgkQdEmbFI1h
Gmx1FA//TwoduTJ9v5wikj1z+ZbY5Q3XLZ6qj/QCNdwFdgcMvRHCv83ZfWLOmtCD
V5DCIuLcyG5st5qOO7+A1R8+Oud5Dql/b3lgPFX/K/BzVoKe3l7aaUVXjSpUHsSu
I7wtdcommE60iNk64D6n1Wf3oiV7s+7lwA4wV8ZDmOmZzfkoM5/Cp4hdREqs6TTJ
4TQvLgHM/gDzXoydndQ3Pun3pImPTmttoKgJUronfqB2gOyflR1N5MvaEBXFsRSy
VhxaZDCtfZfbHB1O+HjaWW9ww2VSLwbmOqFImQpEGI+R3ghhJeZMC0EGmNsDvf3C
6KiCUjV+6JGKUnWyiqty4CQMHEF6VP3OlO+638B5TbJDwpvEmEnlonmx5LiYEqxz
h1RR58X2FGIuDtYmKcyHNwWgdvXY1laH9ao4wOAtc6uiUXOgUdgprSi5ifV7Yhw/
J4JsLkyO+S9N6t8Iu/9MBOmJcBp4rqJBt+7EewaXnUSdaN5u7MJi1g0JAnClBEmH
2iR+Cor58LaCQ6UepKCLh4nkjzVsOnwekd2SoqnmLcXLa/r8HGFQvIAIpwpB9SC1
1/acA+1o8HVJH/b4bj9qBRm0TmV33PM2C7Xy/onykaU/T5k7GsyAj/VjXrgMvVEN
x7pHvjax0j7YsqiCPE9GAELswprcTi1FxhEkIfIUkSb2WxH/cjk=
=jo5e
-----END PGP SIGNATURE-----
